File: Task/exp/python/exp.py
def exp(x:float, times:int = 30)->float:
    """Using Taylor series, exp(x) = Sum ((x^n)/n!)
    """
    result = 0
    factorial = 1 # Avoid repeated calcs
    
    for n in range(times):
        if n > 0:
            factorial *= n
        
        result += (x**n)/factorial
    
    return result

# Testing
print("Self implemented exp: ")
print([exp(x) for x in range(10)])

# Checking
import math
print("\n Built-in exp:")
print([math.exp(x) for x in range(10)])

# OR 

# Precision Checking

print("\nPrecision Check")
for times in range(0, 25, 3): # 0 to 30
    print()
    print([f"{exp(x, times):.6f}" for x in range(5)])

File: Task/exp/C/exp.c
#include <stdio.h>
#include <math.h>

// Function to compute exp(x) using Taylor series
float exp_taylor(float x, int terms) {
    float result = 1.0f;     // x^0 / 0!
    float factorial = 1.0f;
    float power = 1.0f;

    for (int n = 1; n < terms; n++) {
        power *= x;          // x^n
        factorial *= n;      // n!
        result += power / factorial;
    }

    return result;
}

int main(void) {
    printf("Self-implemented exp(x):\n");
    for (int x = 0; x < 10; x++) {
        printf("x = %d: %.6f\n", x, exp_taylor(x, 30));
    }

    printf("\nBuilt-in exp(x):\n");
    for (int x = 0; x < 10; x++) {
        printf("x = %d: %.6f\n", x, exp(x));
    }

    printf("\nPrecision Check:\n");
    for (int terms = 0; terms <= 30; terms += 3) {
        printf("\nterms = %d:\n", terms);
        for (int x = 0; x < 5; x++) {
            printf("exp(%d, %d) = %.6f  ", x, terms, exp_taylor(x, terms));
        }
        printf("\n");
    }

    return 0;
}
